common/lib/common-utils/src/timer.ts

```diff
@@ -57,10 +57,41 @@ interface IRunningTimerState extends ITimeout {
     restart?: ITimeout;
 }
 
+const maxSetTimeoutMs = 0x7FFFFFFF; // setTimeout limit is MAX_INT32=(2^31-1).
+
+/**
+ * Sets timeouts like the setTimeout function allowing timeouts to exceed the setTimeout's max timeout limit.
+ * The longer the timeout, the less accurate the timeout is exactly going to be
+ * @param timeoutFn - executed when the timeout expires
+ * @param timeoutMs - duration of the timeout in milliseconds
+ * @param setTimeoutFn - executed to update the timeout if multiple timeouts are required when timeoutMs > maxTimeout
+ * @returns the initial timeout
+ */
+export function setLongTimeout(
+    timeoutFn: () => void,
+    timeoutMs: number,
+    setTimeoutFn?: (timeout: ReturnType<typeof setTimeout>) => void,
```

Comment:
`setTimeoutFn` made sense to me in my earlier comment but it sounds so similar to `setTimeout` now. Sorry about that.
Maybe `setTimeoutIdFn`? and the parameter / return variable can be `timeoutId`
From https://developer.mozilla.org/en-US/docs/Web/API/setTimeout, the return value is called` timeoutId`